lib/floe/workflow/state.rb

```diff
@@ -19,27 +19,28 @@ def build!(workflow, name, payload)
         end
       end
 
-      attr_reader :workflow, :comment, :name, :type, :payload
+      attr_reader :workflow, :comment, :name, :next_state, :type, :payload
 
       def initialize(workflow, name, payload)
-        @workflow = workflow
-        @name     = name
-        @payload  = payload
-        @end      = !!payload["End"]
-        @type     = payload["Type"]
-        @comment  = payload["Comment"]
+        @workflow   = workflow
+        @name       = name
+        @payload    = payload
+        end_state   = !!payload["End"]
+        @type       = payload["Type"]
+        @comment    = payload["Comment"]
+        @next_state = payload["Next"] unless end_state
```

Comment:
Or perhaps just change this to 

```suggestion
        @next_state = payload["Next"] unless payload["End"]
```